pennylane/interfaces/torch.py

```diff
@@ -246,3 +258,289 @@ def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_d
+            during the forward pass. This function must return a tuple ``(results, jacobians)``.
+            If ``jacobians`` is an empty list, then ``gradient_fn`` is used to
+            compute the gradients during the backwards pass.
+        gradient_kwargs (dict): dictionary of keyword arguments to pass when
+            determining the gradients of tapes
+        gradient_fn (callable): the gradient function to use to compute quantum gradients
+        _n (int): a positive integer used to track nesting of derivatives, for example
+            if the nth-order derivative is requested.
+        max_diff (int): If ``gradient_fn`` is a gradient transform, this option specifies
+            the maximum order of derivatives to support. Increasing this value allows
+            for higher order derivatives to be extracted, at the cost of additional
+            (classical) computational overhead during the backwards pass.
+    Returns:
+        list[list[torch.Tensor]]: A nested list of tape results. Each element in
+        the returned list corresponds in order to the provided tapes.
+    """
+    # pylint: disable=unused-argument
+    parameters = []
+    for tape in tapes:
+        # set the trainable parameters
+        params = tape.get_parameters(trainable_only=False)
+        tape.trainable_params = qml.math.get_trainable_indices(params)
+        parameters.extend(tape.get_parameters())
+
+    kwargs = dict(
+        tapes=tapes,
+        device=device,
+        execute_fn=execute_fn,
+        gradient_fn=gradient_fn,
+        gradient_kwargs=gradient_kwargs,
+        _n=_n,
+        max_diff=max_diff,
+    )
+
+    return ExecuteTapesNew.apply(kwargs, *parameters)
+
+
+def _res_to_torch(r, i, ctx):
+    """Convert results from unwrapped execution to torch."""
+    counts = False
+    if any(
+        m.return_type in (qml.measurements.Counts, qml.measurements.AllCounts)
+        for m in ctx.tapes[i].measurements
+    ):
+        counts = True
+
+    if isinstance(r, (list, tuple)):
+        if counts:
+            res = []
+            for t in r:
+                if isinstance(t, dict):
+                    res.append(t)
+                else:
+                    res.append(torch.as_tensor(t, device=ctx.torch_device))
+        else:
+            res = [torch.as_tensor(t, device=ctx.torch_device) for t in r]
+        if isinstance(r, tuple):
+            res = tuple(res)
+    else:
+        res = torch.as_tensor(r, device=ctx.torch_device)
+
+    return res
+
+
+def _jac_to_torch(i, ctx):
+    """Convert Jacobian from unwrapped execution to torch in the given ctx."""
+    jac_ = []
+    if ctx.jacs:
+        multi_m = len(ctx.tapes[i].measurements) > 1
+        multi_p = len(ctx.tapes[i].trainable_params) > 1
+
+        if multi_p and multi_m:
+            for jac in ctx.jacs[i]:
+                sub_jac = [torch.as_tensor(j, device=ctx.torch_device) for j in jac]

```

Comment:
Minor: slightly adjusting the names `jac_`, `sub_jac` and `jac` do be more descriptive could improve readability around here, I think right now one has to keep in mind which object does what.